src/actions/activities.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Define the Temporal worker activities actions."""
+
+# pylint: disable=W0212,R0801
+
+import logging
+
+from ops import framework
+
+from log import log_event_handler
+
+logger = logging.getLogger(__name__)
+
+
+class ActivitiesActions(framework.Object):
+    """Client for Temporal worker activities actions."""
+
+    def __init__(self, charm):
+        """Construct.
+
+        Args:
+            charm: The charm to attach the hooks to.
+        """
+        super().__init__(charm, "activities-actions")
+        self.charm = charm
+        charm.framework.observe(charm.on.add_activities_action, self._on_add_activities_action)
+        charm.framework.observe(charm.on.remove_activities_action, self._on_remove_activities_action)
+        charm.framework.observe(charm.on.list_activities_action, self._on_list_activities_action)
+
+    @log_event_handler(logger)
+    def _on_add_activities_action(self, event):
+        """Add to the list of activities supported by worker.
+
+        Args:
+            event: The event triggered when the action is triggered.
+        """
+        if not self.charm._state.is_ready():
+            event.defer()
+            return
+
+        if not self.charm.unit.is_leader():
+            event.fail("action cannot be performed on non-leader unit")
+            return
+
+        sw = self.charm._state.supported_activities
+        for workflow in event.params["activities"].split(","):
```

Comment:
Is there a chance that there is confusion from someone putting in `a, b, c` rather than `a,b,c` here? I did not see if you validate or strip this somewhere else.

There is similar usage in other places as well.

PS. After looking through the entire PR I guess you use `split(",")` everywhere so you would not have problems with strings not matching. Still, if this string ever faces the user (e.g. in the logs) it would be nicer to trim the whitespace, What do you think?

Reply:
You're right, i'll trim the whitespace and add some minor validation to the user input in the next PR when I add the other charm actions.